packages/docusaurus/src/commands/deploy.ts

```diff
@@ -169,42 +169,35 @@ Try using DEPLOYMENT_BRANCH=main or DEPLOYMENT_BRANCH=master`);
     );
     if (
       shellExecLog(
-        `git clone --depth 1 --no-single-branch ${remoteBranch} ${toPath}`,
+        `git clone --depth 1 --branch ${deploymentBranch} ${remoteBranch} ${toPath}`,
       ).code !== 0
     ) {
-      throw new Error(`Running "git clone" command in "${toPath}" failed.`);
-    }
+      if (
+        shellExecLog(`git clone --depth 1 ${remoteBranch} ${toPath}`).code !== 0
+      ) {
+        throw new Error(`Running "git clone" command in "${toPath}" failed.`);
+      }
 
-    shell.cd(toPath);
+      shell.cd(toPath);
 
-    // If the default branch is the one we're deploying to, then we'll fail
-    // to create it. This is the case of a cross-repo publish, where we clone
-    // a github.io repo with a default branch.
-    const defaultBranch = shell
-      .exec('git rev-parse --abbrev-ref HEAD')
-      .stdout.trim();
-    if (defaultBranch !== deploymentBranch) {
-      if (shellExecLog(`git checkout origin/${deploymentBranch}`).code !== 0) {
+      // If the default branch is the one we're deploying to, then we'll fail
+      // to create it. This is the case of a cross-repo publish, where we clone
+      // a github.io repo with a default branch.
+      const defaultBranch = shell
+        .exec('git rev-parse --abbrev-ref HEAD')
+        .stdout.trim();
+      if (defaultBranch !== deploymentBranch) {
         if (
           shellExecLog(`git checkout --orphan ${deploymentBranch}`).code !== 0
         ) {
           throw new Error(
             `Running "git checkout ${deploymentBranch}" command failed.`,
           );
         }
-      } else if (
-        shellExecLog(`git checkout -b ${deploymentBranch}`).code +
-          shellExecLog(
-            `git branch --set-upstream-to=origin/${deploymentBranch}`,
-          ).code !==
-        0
-      ) {
-        throw new Error(
-          `Running "git checkout ${deploymentBranch}" command failed.`,
-        );
       }
     }
 
+    shell.cd(toPath);
```

Comment:
Can you explain why this needs to be moved?

Reply:
The current solution calls this command twice. See: https://github.com/facebook/docusaurus/blob/c541e2d83c0066e92dd13f018698081bf78b387c/packages/docusaurus/src/commands/deploy.ts#L183-L221

The second call seems to be redundant as the directory is not changed between the first call and the second call. So in the current code, the second `shell.cd` call can be removed.

But with the current refactoring, the first `shell.cd` is only called before checking out an orphan branch. So to ensure that we're in the correct directory before calling `git rm -rf .`, we'll need to move the command.

If we're going to go with the additional refactoring I've suggested in the previous comment, this will be moved even further up, into an `if`-clause.